File: warrior/systems/SoundEffectSystem.java
package net.brokenspork.systems;

import net.brokenspork.components.SoundEffect;
import net.brokenspork.components.SoundEffect.EFFECT;

import com.artemis.Aspect;
import com.artemis.ComponentMapper;
import com.artemis.Entity;
import com.artemis.annotations.Mapper;
import com.artemis.systems.EntityProcessingSystem;
import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.audio.Sound;

public class SoundEffectSystem extends EntityProcessingSystem {

	@Mapper
	ComponentMapper<SoundEffect> se;

	Sound pew = Gdx.audio.newSound(Gdx.files.internal("sounds/pew.wav"));
	Sound asplode = Gdx.audio.newSound(Gdx.files.internal("sounds/asplode.wav"));
	Sound smallasplode = Gdx.audio.newSound(Gdx.files.internal("sounds/smallasplode.wav"));

	@Override
	protected void initialize() {

	}

	@SuppressWarnings("unchecked")
	public SoundEffectSystem() {
		super(Aspect.getAspectForAll(SoundEffect.class));
	}

	@Override
    protected void process(Entity e) {

		SoundEffect soundEffect = se.get(e);
		
		switch (soundEffect.effect) {
		case PEW:
			pew.play();
			break;
		case ASPLODE:
			asplode.play();
			break;
		case SMALLASPLODE:
			smallasplode.play();
			break;
		default:
			break;
		}

		e.removeComponent(soundEffect);
		e.changedInWorld();
    }
}

File: cocos/systems/SoundEffectSystem.ts
module example.systems {
	
	import SoundEffect = example.components.SoundEffect;
	import EFFECT = example.components.EFFECT;
	
	import Aspect = artemis.Aspect;
	import ComponentMapper = artemis.ComponentMapper;
	import Entity = artemis.Entity;
	import EntityProcessingSystem = artemis.systems.EntityProcessingSystem;
	import Mapper = artemis.annotations.Mapper;
	// import  = badlogic.gdx.Gdx;
	// import  = badlogic.gdx.audio.Sound;
	
	export class SoundEffectSystem extends EntityProcessingSystem {
	
		@Mapper(SoundEffect) se:ComponentMapper<SoundEffect>;
	
		// pew:Sound = Gdx.audio.newSound(Gdx.files.internal("sounds/pew.wav"));
		// asplode:Sound = Gdx.audio.newSound(Gdx.files.internal("sounds/asplode.wav"));
		// smallasplode:Sound = Gdx.audio.newSound(Gdx.files.internal("sounds/smallasplode.wav"));
	
		
		public initialize() {
	
		}
	
		//@SuppressWarnings("unchecked")
		constructor() {
			super(Aspect.getAspectForAll(SoundEffect));
		}
	
		
		public processEach(e:Entity) {
	
			var soundEffect:SoundEffect = this.se.get(e);

			switch (soundEffect.effect) {
			case EFFECT.PEW:
				//pew.play();
				break;
			case EFFECT.ASPLODE:
				//asplode.play();
				break;
			case EFFECT.SMALLASPLODE:
				//smallasplode.play();
				break;
			default:
				break;
			}
	
			e.removeComponentInstance(soundEffect);
			e.changedInWorld();
		}
	}
}

